ignite/contrib/metrics/cohen_kappa.py

```diff
@@ -0,0 +1,72 @@
+from typing import Callable, Optional, Union
+
+import torch
+
+from ignite.metrics import EpochMetric
+
+
+class CohenKappa(EpochMetric):
+    """Compute different types of Cohen's Kappa: Non-Wieghted, Linear, Quadratic.
+    Accumulating predictions and the ground-truth during an epoch and applying
+    `sklearn.metrics.cohen_kappa_score <https://scikit-learn.org/stable/modules/
+    generated/sklearn.metrics.cohen_kappa_score.html>`_ .
+
+    Args:
+        output_transform: a callable that is used to transform the
+            :class:`~ignite.engine.engine.Engine`'s ``process_function``'s output into the
+            form expected by the metric. This can be useful if, for example, you have a multi-output model and
+            you want to compute the metric with respect to one of the outputs.
+        weights: a string is used to define the type of Cohen's Kappa whether Non-Weighted or Linear or Quadratic
+            (default: "")
+        check_compute_fn: Default False. If True, `cohen_kappa_score
+            <https://scikit-learn.org/stable/modules/generated/sklearn.metrics.cohen_kappa_score.html>`_
+            is run on the first batch of data to ensure there are
+            no issues. User will be warned in case there are any issues computing the function.
+        device: optional device specification for internal storage.
+
+    .. code-block:: python
+
+        def activated_output_transform(output):
+            y_pred, y = output
+            return y_pred, y
+
+        weights = None or linear or quadratic
+
+        cohen_kappa = CohenKappa(activated_output_transform, weights)
+
+    """
+
+    def __init__(
+        self,
+        output_transform: Callable = lambda x: x,
+        weights: Optional[str] = None,
+        check_compute_fn: bool = False,
+        device: Union[str, torch.device] = torch.device("cpu"),
+    ):
+
+        try:
+            from sklearn.metrics import cohen_kappa_score
+        except ImportError:
+            raise RuntimeError("This contrib module requires sklearn to be installed.")
+
+        # initalize weights
+        self.weights = weights
```

Comment:
Shouldn't we check provided `weights` values ? Like if user mispell the name etc.

Reply:
But that's already provided in sklearn 
`
ValueError: Unknown kappa weighting type.
`

Reply:
Yes, but user will get this error in the end of the epoch (after a certain time) if not using `check_compute_fn=True` which is by default `False`.

Reply:
Okay i will add the checks and the `pytest.approx`

Reply:
And I forgot to mention that we have to run a check for that to ensure better codecov. Search for `with pytest.raises(...):` in the codebase to see how to do that.

Reply:
Well it seems it's the only error in the CI we got not, i will do it.

Reply:
@vfdev-5, all updated.